### PR TITLE
Better sidenotes with Prince sidenotes area

### PR DESCRIPTION
**Pull request by arthurattwell**
This includes the work in #662 supporting Prince for Books. There, I noted:

>In addition to some CSS fixes implemented during testing, this supports and implements using Prince for Books as the default Prince version for PDF output. This is associated with https://github.com/electricbookworks/node-prince/pull/4.
>
> One important change here is that we now pass the book's stylesheet to Prince as a 'user stylesheet'. The advantage of this is that Prince will apply the stylesheet to any SVGs referenced in the document with `img src="*.svg"`. In previous Prince versions, it was already the case that the book's CSS would apply to SVGs. This had the advantage of letting us apply the parent document's fonts to SVGs. The potential downside of loading the styles as a user stylesheet is that this appears to replace (not just 'extend') the CSS loaded by `style` tags in the HTML. For our use case, this is fine since we only ever load one stylesheet, but in theory one could reference multiple stylesheets in HTML, which may then be ignored when we load a user stylesheet. I think this is worth it, given the importance of loading fonts correctly in SVGs, without having to define font styles and paths to font files in individual SVG files.
>
> The current version of Prince for Books appears to be based on Prince 14. We may soon switch to Prince 15 instead, given its [support for a proper @sidenotes area](https://css4.pub/2023/sidenotes/). There is not yet a Prince for Books based on Prince 15.

As it happens, the current Prince for Books also supports [sidenotes](https://css4.pub/2023/sidenotes/), which is great!

This PR then implements that sidenotes area. At the same time, I've done a bit of housekeeping:

- replaced the old sidebar technique with the new sidenotes area
- align margin-box content (headers, footers) with sidebar
- check that screen PDF inherits all print-pdf metadata by default (including TOC and files)
- turn sidebar on by default, and use a larger page size for Samples
- lint all JS for Standard JS
- rename 'print' to 'pdf' in all partials (since screen PDF is not 'print')

### CODE REVIEW

File: _sass/template/partials/_pdf-slides.scss [hunk 1]
```diff
@@ -193,23 +193,25 @@ $print-slides: true !default;
...
+                display: inline-block;
+                box-sizing: border-box;
 
                 // When there are only two slides (a summary and two figures)
                 // we shift those inside to avoid the empty third column.
                 // This CSS selects the second and third of three siblings.
                 &:nth-child(2):nth-last-child(2),
                 &:nth-child(3):nth-last-child(1) {
-                    margin-outside: $sidebar-text-width + $sidebar-margin-inside;
+                    // margin-outside: $sidebar-content-width + $sidebar-gap-width;
```
Comment (LouiseSteward) [inline _sass/template/partials/_pdf-slides.scss:214]: If this is commented out, the selector is empty - can we remove it?

File: _sass/template/print-pdf.scss [hunk 2]
```diff
@@ -380,72 +381,75 @@ $print-page-setup-lightning-source: false !default; // Removes crop marks and se
...
+@import "template/partials/pdf-epigraphs";
+@import "template/partials/pdf-dedications";
+@import "template/partials/pdf-glossaries"; // Including .glossary
+@import "template/partials/pdf-highlighter";
+@import "template/partials/pdf-pullquotes"; // Supports .pullquote
+@import "template/partials/pdf-boxes";
+@import "template/partials/pdf-code"; // For computer code and preformatted text
+@import "template/partials/pdf-tables";
+@import "template/partials/pdf-figures";
+@import "template/partials/pdf-notes"; // Footnotes, endnotes, sidenotes // CRASHES PRINCE
```
Comment (LouiseSteward) [inline _sass/template/print-pdf.scss:430]: Do we need to add more detail (either here, or in pdf-notes.scss) about what might lead to this crash?

File: assets/js/accordion.js [hunk 3]
```diff
@@ -1,6 +1,5 @@
-/*jslint browser, for */
-/*global window, ebLazyLoadImages, searchTerm, videoShow
-    locales, pageLanguage, console, Element, HTMLDocument,
+/* global window, ebLazyLoadImages, videoShow
```
Comment (LouiseSteward) [inline assets/js/accordion.js:1]: Are all of the changes to the JS just linting, or are there code changes that need closer review?

Comment (arthurattwell) [reply assets/js/accordion.js:1]: Definitely just linting.

File: _sass/template/print-pdf.scss (hunk 2 above)
Comment (arthurattwell) [reply _sass/template/print-pdf.scss:430]: Thanks, this was resolved, so I'll remove that debugging note. (If a single sidenote is longer than can fit on one page, Prince crashes. So it's caused by content, not CSS.)

File: _sass/template/partials/_pdf-slides.scss (hunk 1 above)
Comment (arthurattwell) [reply _sass/template/partials/_pdf-slides.scss:214]: Hmm, actually I need to do more testing here before we merge.

File: _sass/template/print-pdf.scss (hunk 2 above)
Comment (arthurattwell) [reply _sass/template/print-pdf.scss]: Done in [1f2fa57](https://github.com/electricbookworks/electric-book/pull/664/commits/1f2fa57d8a08924cbbae47e1f99205205af7c3a4).

File: _sass/template/partials/_pdf-slides.scss (hunk 1 above)
Comment (arthurattwell) [reply _sass/template/partials/_pdf-slides.scss]: After more testing, this and other issues resolved in [13ab587](https://github.com/electricbookworks/electric-book/pull/664/commits/13ab5875302d269f831aa31761852088128bd3fe).

**Review by LouiseSteward — APPROVED** [review]

**Pull request merged**